src/win.c

```diff
@@ -2254,48 +2290,16 @@ void map_win_start(session_t *ps, struct managed_win *w) {
 	// XXX Can we assume map_state is always viewable?
 	w->a.map_state = XCB_MAP_STATE_VIEWABLE;
 
-	win_update_screen(ps->xinerama_nscrs, ps->xinerama_scr_regs, w);
-
-	// Set window event mask before reading properties so that no property
-	// changes are lost
-	xcb_change_window_attributes(
-	    ps->c, w->base.id, XCB_CW_EVENT_MASK,
-	    (const uint32_t[]){determine_evmask(ps, w->base.id, WIN_EVMODE_FRAME)});
-
-	// Get notification when the shape of a window changes
-	if (ps->shape_exists) {
-		xcb_shape_select_input(ps->c, w->base.id, 1);
-	}
-
 	// Update window mode here to check for ARGB windows
 	w->mode = win_calc_mode(w);
 
-	// Detect client window here instead of in add_win() as the client
-	// window should have been prepared at this point
-	if (!w->client_win) {
-		win_recheck_client(ps, w);
-	} else {
-		// Re-mark client window here
-		win_mark_client(ps, w, w->client_win);
-	}
-	assert(w->client_win);
-
 	log_debug("Window (%#010x) has type %s", w->base.id, WINTYPES[w->window_type]);
 
 	// TODO(yshui) can we just replace calls below with win_on_factor_change?
 
 	// Update window focus state
 	win_update_focused(ps, w);
```

Comment:
Do we have to run all the logic for `win_on_factor_change` when we map the window? Or can we rely on `recheck_focus()` being called in the critical section to take care of that?

Reply:
I don't want to change that in this PR. We can break it up later if we want to.